src/main/java/se/bjurr/violations/lib/example/MyClass.java

```diff
@@ -30,6 +31,6 @@ void npe2(String a, String b) {
 
  @Override
  public boolean equals(Object obj) {
-  return true;
+  return false;
  }
```

Comment:
**Reporter**: Findbugs
**Rule**: HE_EQUALS_USE_HASHCODE
**Severity**: WARN
**File**: se/bjurr/violations/lib/example/MyClass.java L35
**Source**: se.bjurr.violations.lib.example.MyClass

Class defines equals() and uses Object.hashCode()&#10;&#10;&#10;  &lt;p&gt; This class overrides &lt;code&gt;equals(Object)&lt;/code&gt;, but does not&#10;  override &lt;code&gt;hashCode()&lt;/code&gt;, and inherits the implementation of&#10;  &lt;code&gt;hashCode()&lt;/code&gt; from &lt;code&gt;java.lang.Object&lt;/code&gt; (which returns&#10;  the identity hash code, an arbitrary value assigned to the object&#10;  by the VM).&amp;nbsp; Therefore, the class is very likely to violate the&#10;  invariant that equal objects must have equal hashcodes.&lt;/p&gt;&#10;&#10;&lt;p&gt;If you don&#39;t think instances of this class will ever be inserted into a HashMap/HashTable,&#10;the recommended &lt;code&gt;hashCode&lt;/code&gt; implementation to use is:&lt;/p&gt;&#10;&lt;pre&gt;public int hashCode() {&#10;  assert false : &quot;hashCode not designed&quot;;&#10;  return 42; // any arbitrary constant will do&#10;  }&lt;/pre&gt;&#10;


*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-2046975042>*